crucible/src/Lang/Crucible/Simulator/SymSequence.hs

```diff
@@ -0,0 +1,302 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE KindSignatures #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeOperators #-}
+
+module Lang.Crucible.Simulator.SymSequence
+( SymSequence(..)
+, nilSymSequence
+, consSymSequence
+, appendSymSequence
+, muxSymSequence
+, isNilSymSequence
+, lengthSymSequence
+, headSymSequence
+, tailSymSequence
+, unconsSymSequence
+) where
+
+import           Data.Functor.Const
+import           Data.Kind (Type)
+import           Data.IORef
+import           Data.Maybe (isJust)
+import           Data.Parameterized.Nonce
+import qualified Data.Parameterized.Map as MapF
+
+import           Lang.Crucible.Types
+import           What4.Interface
+import           What4.Partial
+
+------------------------------------------------------------------------
+-- SymSequence
+
+data SymSequence sym a where
```

Comment:
It might be worth giving this a `Show`/`ShowF`/`Pretty` instance to make it easier to print these for debugging purposes in the future.

Reply:
agreed